frappe/core/doctype/file/file.py

```diff
@@ -664,20 +667,30 @@ def get_web_image(file_url):
 	return image, filename, extn
 
 
-def delete_file(path):
+def delete_file(path, trash=False):
 	"""Delete file from `public folder`"""
 	if path:
 		if ".." in path.split("/"):
 			frappe.msgprint(_("It is risky to delete this file: {0}. Please contact your System Manager.").format(path))
 
 		parts = os.path.split(path.strip("/"))
-		if parts[0]=="files":
+		is_private = False if parts[0]=="files" else True
+		if not is_private:
 			path = frappe.utils.get_site_path("public", "files", parts[-1])
 
 		else:
 			path = frappe.utils.get_site_path("private", "files", parts[-1])
 
 		path = encode(path)
+		if trash:
+			trash_dir = frappe.get_site_path("private" if is_private else "public", "files", ".trash")
+
+			if not os.path.exists(trash_dir):
+				os.mkdir(trash_dir)
+			
+			if os.path.exists(path):
+				shutil.copy2(path, '/'.join([trash_dir, parts[-1]]))
```

Comment:
- Use `os.path.join` to generate a file path
- Assign parts[-1] to a variable which makes it easier to understand its the file name; or use `os.path.basename` to retrieve it from path